### ISSUE TITLE
make max-len ignore rule disabling lines

### ISSUE THREAD

**Issue by rafaeltab** [eslint/eslint#14836]
**What rule do you want to change?**
`max-len`
**Does this change cause the rule to produce more or fewer warnings?**
fewer
**How will the change be implemented? (New option, new default behavior, etc.)?**
Could be either,
an option with the name `ignoreDisableComments` could be created, or it could simply be added as default behavior, since I don't think anyone really wants this rule to mark rule disables as a problem anyway.

**Please provide some example code that this change will affect:**
```js
// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
createdObj[currentObject] = val;
```

**What does the rule currently do for this code?**
This currently triggers `max-len` and creates an error if you've set `max-len` to 100 without disabling all comments

**What will the rule do after it's changed?**
It will no longer flag it as to long.

**Are you willing to submit a pull request to implement this change?**
Sorry, again, I have no clue how your code base is built.

**Current workarounds (Added this header myself)**
You can turn the `ignoreComments` option on, and it will no longer mark this as a problem. However, some people do want this rule for regular comments that explain the code, but not for comments that disable a rule. It's a bit of a niche, but I was still surprised this wasn't a thing yet.

Also, you can disable the rule for this rule disabling line as follows, this does work
```ts
// eslint-disable-next-line max-len
// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
createdObj[currentObject] = val;
```

I still think it should be default behavior to not mark this as an error or warning though.

**Comment by mdjermanovic** [comment]
> an option with the name `ignoreDisableComments` could be created, or it could simply be added as default behavior, since I don't think anyone really wants this rule to mark rule disables as a problem anyway.

Currently, `eslint-disable-next-line` directive comments must be on a single line, so it does make sense to ignore them by default in `max-len`, as there is no way to fix the code. However, if we allow multiline `eslint-disable-next-line` comments (as proposed in https://github.com/eslint/eslint/issues/14284), ignoring them might be undesirable for some users.

I'm not opposed to adding a new option, but this can be already achieved with the existing [ignorePattern](https://eslint.org/docs/rules/max-len#ignorepattern) option. [Here's a demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG1heC1sZW46IFtcImVycm9yXCIsIDgwLCB7XG4gICAgaWdub3JlUGF0dGVybjogXCJeXFxcXHMqWy9dWy8qXVxcXFxzKmVzbGludC1kaXNhYmxlLW5leHQtbGluZVwiIFxufV0gKi9cblxuLy8gZXNsaW50LWRpc2FibGUtbmV4dC1saW5lIG5vLXVuZGVmLCBuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmcsIG5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3JcblxuLyogZXNsaW50LWRpc2FibGUtbmV4dC1saW5lIG5vLXVuZGVmLCBuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmcsIG5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IgKi9cblxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJuby11bmRlZiI6Mn0sImVudiI6e319fQ==) with a regular expression that matches with those lines. Would that work for you?

**Comment by rafaeltab** [comment]
> I'm not opposed to adding a new option, but this can be already achieved with the existing [ignorePattern](https://eslint.org/docs/rules/max-len#ignorepattern) option. [Here's a demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG1heC1sZW46IFtcImVycm9yXCIsIDgwLCB7XG4gICAgaWdub3JlUGF0dGVybjogXCJeXFxcXHMqWy9dWy8qXVxcXFxzKmVzbGludC1kaXNhYmxlLW5leHQtbGluZVwiIFxufV0gKi9cblxuLy8gZXNsaW50LWRpc2FibGUtbmV4dC1saW5lIG5vLXVuZGVmLCBuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmcsIG5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3JcblxuLyogZXNsaW50LWRpc2FibGUtbmV4dC1saW5lIG5vLXVuZGVmLCBuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmcsIG5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IgKi9cblxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJuby11bmRlZiI6Mn0sImVudiI6e319fQ==) with a regular expression that matches with those lines. Would that work for you?

This works perfectly fine! I'll add it to my project right away.

It isn't the most elegant method, so it would still be better to add an actual option. However, this makes it even less necessary then before.

**Comment by rafaeltab** [comment]
Closing since a resolution has been reached. A little tl;dr; for people trying to find what they could(/should?) use is below.

TL;DR; There is no such option that disables the disabling lines. However, you can use ignorePattern like used in mdjermanovic's demo regex.
